tst/EnergyPlus/unit/DaylightingManager.unit.cc

```diff
@@ -1395,3 +1398,274 @@ TEST_F(EnergyPlusFixture, DaylightingManager_DayltgInteriorIllum_Test)
+        "  0.8;                     !- Z-Coordinate of Reference Point {m}",
+
+    });
+
+    ASSERT_TRUE(process_idf(idf_objects));
+
+    bool foundErrors = false;
+    HeatBalanceManager::GetZoneData(foundErrors);
+    ASSERT_FALSE(foundErrors);
+
+    int numObjs = inputProcessor->getNumObjectsFound("Daylighting:Controls");
+    EXPECT_EQ(1, numObjs);
+
+    DaylightingManager::GetInputDayliteRefPt(foundErrors);
+    compare_err_stream("");
+    EXPECT_FALSE(foundErrors);
+    EXPECT_EQ(10, DataDaylighting::TotRefPoints);
+
+    DaylightingManager::GetDaylightingControls(numObjs, foundErrors);
+    // Used to throw
+    //    ** Severe  ** GetDaylightingControls: Fraction of Zone controlled by the Daylighting reference points is > 1.0.
+    //    **   ~~~   ** ..discovered in \"Daylighting:Controls\" for Zone=\"WEST ZONE\", trying to control 1.00 of the zone.\n
+    compare_err_stream("");
+    EXPECT_FALSE(foundErrors);
+
+    EXPECT_EQ("WEST ZONE_DAYLCTRL", DataDaylighting::ZoneDaylight(1).Name);
+    EXPECT_EQ("WEST ZONE", DataDaylighting::ZoneDaylight(1).ZoneName);
+    EXPECT_EQ(DataDaylighting::SplitFluxDaylighting, DataDaylighting::ZoneDaylight(1).DaylightMethod);
+    EXPECT_EQ(DataDaylighting::Continuous, DataDaylighting::ZoneDaylight(1).LightControlType);
+
+    EXPECT_EQ(0.3, DataDaylighting::ZoneDaylight(1).MinPowerFraction);
+    EXPECT_EQ(0.2, DataDaylighting::ZoneDaylight(1).MinLightFraction);
+    EXPECT_EQ(1, DataDaylighting::ZoneDaylight(1).LightControlSteps);
+    EXPECT_EQ(1.0, DataDaylighting::ZoneDaylight(1).LightControlProbability);
+
+    EXPECT_EQ(1, DataDaylighting::ZoneDaylight(1).glareRefPtNumber);
+    EXPECT_EQ(180., DataDaylighting::ZoneDaylight(1).ViewAzimuthForGlare);
+    EXPECT_EQ(20., DataDaylighting::ZoneDaylight(1).MaxGlareallowed);
+    EXPECT_EQ(0, DataDaylighting::ZoneDaylight(1).DElightGriddingResolution);
+
+    EXPECT_EQ(10, DataDaylighting::ZoneDaylight(1).TotalDaylRefPoints);
+
+    std::array<Real64, 10> fractions({0.1053, 0.0936, 0.1213, 0.1018, 0.0893, 0.0842, 0.0882, 0.1026, 0.1134, 0.1003});
+    Real64 sum(0.0);
+    int i = 1;
+    for (auto frac: fractions) {
+        sum += frac;
+        EXPECT_EQ(i, DataDaylighting::ZoneDaylight(1).DaylRefPtNum(i));
+        EXPECT_EQ("WEST ZONE_DAYLREFPT" + std::to_string(i),
+                  DataDaylighting::DaylRefPt(DataDaylighting::ZoneDaylight(1).DaylRefPtNum(i)).Name);
+        EXPECT_EQ(frac, DataDaylighting::ZoneDaylight(1).FracZoneDaylit(i));
+        EXPECT_EQ(200., DataDaylighting::ZoneDaylight(1).IllumSetPoint(i));
+        ++i;
+    }
+
+    // It does sum up to 1.0
+    EXPECT_DOUBLE_EQ(1.0, sum);
+    EXPECT_FALSE(std::abs(sum - 1.0) > std::numeric_limits<double>::epsilon());
+    // Yet, if you are being very litteral, then it's slightly more
+    EXPECT_TRUE(sum > 1.0);
+    EXPECT_FALSE(sum < 1.0);
+
+}
+
+TEST_F(EnergyPlusFixture, DaylightingManager_GetInputDaylightingControls_NotAroundOne)
```

Comment:
And this one clearly demonstrates the proper warning message with the extra digit printed.  Good stuff.